dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/TagApi.java

```diff
@@ -0,0 +1,192 @@
+import org.jclouds.collect.internal.ArgsToPagedIterable;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.domain.PaginatedCollection;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Tag;
+import org.jclouds.dimensiondata.cloudcontrol.domain.TagInfo;
+import org.jclouds.dimensiondata.cloudcontrol.domain.TagKey;
+import org.jclouds.dimensiondata.cloudcontrol.domain.TagKeys;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Tags;
+import org.jclouds.dimensiondata.cloudcontrol.filters.OrganisationIdFilter;
+import org.jclouds.dimensiondata.cloudcontrol.options.PaginationOptions;
+import org.jclouds.dimensiondata.cloudcontrol.utils.ParseResponse;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@RequestFilters({ BasicAuthentication.class, OrganisationIdFilter.class })
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/{jclouds.api-version}/tag")
+public interface TagApi {
+
+   @Named("tag:createTagKey")
+   @POST
+   @Path("/createTagKey")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @ResponseParser(TagKeyId.class)
+   String createTagKey(@PayloadParam("name") String name, @PayloadParam("description") String description,
+         @PayloadParam("valueRequired") Boolean valueRequired,
+         @PayloadParam("displayOnReport") Boolean displayOnReport);
+
+   @Named("tag:applyTags")
+   @POST
+   @Path("/applyTags")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @MapBinder(BindToJsonPayload.class)
+   void applyTags(@PayloadParam("assetId") String assetId, @PayloadParam("assetType") String assetType,
+         @PayloadParam("tagById") List<TagInfo> tagById);
+
+   @Named("tag:removeTags")
+   @POST
+   @Path("/removeTags")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @MapBinder(BindToJsonPayload.class)
+   void removeTags(@PayloadParam("assetId") String assetId, @PayloadParam("assetType") String assetType,
+         @PayloadParam("tagKeyId") List<String> tagKeyName);
+
+   @Named("tag:tagKey")
+   @GET
+   @Path("/tagKey")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @ResponseParser(ParseTagKeys.class)
+   PaginatedCollection<TagKey> listTagKeys(PaginationOptions options);
```

Comment:
Add the method without parameters that returns the `PagedIterable` with all keys.